### PR TITLE
CLI skeleton support for FDB Mac

### PR DESCRIPTION
**Pull request by ViramShah91**
sonic-update-1
sonic-update-2

### CODE REVIEW

File: src/CLI/actioner/sonic-cli-mac.py
Comment (justinejose91) [inline src/CLI/actioner/sonic-cli-mac.py]: This function is not needed, as this sorts the interface name, or we may need to extend this function inorder to sort the VLAN names. Need to move this function to a new file, sonic_cli_utils.py.

File: src/CLI/clitree/cli-xml/mac.xml
Comment (justinejose91) [inline src/CLI/clitree/cli-xml/mac.xml:47]: Better to have the command as #show MAC address-table and "address", "count" and "vlan" as PARAMs. So, this can be extended in future, so that you don't need to write separate command for all the sub-commands of MAC. Please hold on to this comment, let's discuss with joyas on this, and make a final decision, since this makes action unique, rather than having a logic for ACTION script.

File: src/CLI/clitree/cli-xml/mac.xml
Comment (justinejose91) [inline src/CLI/clitree/cli-xml/mac.xml]: Do we need this optional flag?

File: src/CLI/clitree/cli-xml/mac.xml
Comment (justinejose91) [inline src/CLI/clitree/cli-xml/mac.xml]: Please fix indentation issues in the code.

File: src/CLI/clitree/cli-xml/mac.xml
Comment (justinejose91) [inline src/CLI/clitree/cli-xml/mac.xml]: Better to have help string start with CAPS, and do we need to mention static in help string? If its not important, can you please check other places too.

File: src/CLI/clitree/cli-xml/mac.xml
Comment (justinejose91) [inline src/CLI/clitree/cli-xml/mac.xml]: Can we change PTYPE as LAG_ID? and in line number 303.

File: src/CLI/clitree/cli-xml/mac.xml
Comment (justinejose91) [inline src/CLI/clitree/cli-xml/mac.xml]: Does it have to start with CAPS for "Address", "Interface" and "Vlan" subcommands? We use small letters in os10. Please correct me if I am wrong.

File: src/CLI/clitree/cli-xml/mac.xml
Comment (justinejose91) [inline src/CLI/clitree/cli-xml/mac.xml]: Please correct the VLAN ID range, and line number 163.

File: src/CLI/clitree/cli-xml/mac.xml
Comment (ViramShah91) [reply src/CLI/clitree/cli-xml/mac.xml]: The requirement is to get result for "show mac address-table static" as well as for "show mac address-table static {vlan | interface | address}". Without optional param,  CLI will be complaining as an incomplete command and we won't get result for static.

File: src/CLI/actioner/sonic-cli-mac.py
Comment (ViramShah91) [reply src/CLI/actioner/sonic-cli-mac.py]: Removed code from the file.

File: src/CLI/clitree/cli-xml/mac.xml
Comment (ViramShah91) [reply src/CLI/clitree/cli-xml/mac.xml]: Fixed indentation.

File: src/CLI/clitree/cli-xml/mac.xml
Comment (ViramShah91) [reply src/CLI/clitree/cli-xml/mac.xml]: Addressed this issue in latest commit.

File: src/CLI/clitree/cli-xml/mac.xml
Comment (ViramShah91) [reply src/CLI/clitree/cli-xml/mac.xml]: Added LAG_ID in sonic-type and updated ptype for port-channel as LAG_ID.

File: src/CLI/clitree/cli-xml/mac.xml
Comment (ViramShah91) [reply src/CLI/clitree/cli-xml/mac.xml]: Addressed this issue in latest commit.

File: src/CLI/clitree/cli-xml/mac.xml
Comment (ViramShah91) [reply src/CLI/clitree/cli-xml/mac.xml]: Addressed this issue in the latest commit.

File: src/CLI/clitree/cli-xml/mac.xml
Comment (justinejose91) [inline src/CLI/clitree/cli-xml/mac.xml]: Could we represent physical interface just like how it's been represented in Kernel? For example: ethernet -> Ethernet, and vlan -> Vlan, port-channel should be PortChannel?

Comment (ViramShah91) [reply src/CLI/clitree/cli-xml/mac.xml]: Addressed.

**Review by justinejose91 — APPROVED** [review]

**Comment by jeffyin-dell** [comment]
@ViramShah91 -- can't merge until your resolve the conflicts in `sonic_types.xml`

**Review by arunsundark — APPROVED** [review]

**Review by jeffyin-dell — APPROVED** [review]

**Pull request merged**
